zebra-chain/src/orchard/sinsemilla.rs

```diff
@@ -45,7 +45,7 @@ pub fn extract_p_bottom(maybe_point: Option<pallas::Point>) -> Option<pallas::Ba
 /// <https://zips.z.cash/protocol/nu5.pdf#concretegrouphashpallasandvesta>
 #[allow(non_snake_case)]
 pub fn pallas_group_hash(D: &[u8], M: &[u8]) -> pallas::Point {
-    let domain_separator = std::str::from_utf8(D).unwrap();
+    let domain_separator = std::str::from_utf8(D).expect("slice is valid utf-8");
```

Comment:
Can we be more specific here?

The domain separators are constant strings, which are guaranteed to be valid UTF-8 by the protocol. The message is also a slice, and it can be arbitrary data.

```suggestion
    let domain_separator = std::str::from_utf8(D).expect("domain separator is valid utf-8");
```